nsxt/resource_nsxt_logical_router_uplink_port.go

```diff
@@ -0,0 +1,237 @@
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceNsxtLogicalRouterUpLinkPortCreate(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(nsxtClients).NsxtClient
+	if nsxClient == nil {
+		return resourceNotSupportedError()
+	}
+
+	description := d.Get("description").(string)
+	displayName := d.Get("display_name").(string)
+	tag := getTagsFromSchema(d)
+	logicalRouterID := d.Get("logical_router_id").(string)
+	linkedLogicalSwitchPortID := d.Get("linked_logical_switch_port_id").(string)
+	edgeClusterMemberIndex := getInt64ListFromSchemaList(d, "edge_cluster_member_index")
+	subnets := getIPSubnetsFromSchema(d)
+	urpfMode := d.Get("urpf_mode").(string)
+	macAddress := d.Get("mac_address").(string)
+	logicalRouterUpLinkPort := manager.LogicalRouterUpLinkPort{
+		Description:               description,
+		DisplayName:               displayName,
+		Tags:                      tag,
+		LogicalRouterId:           logicalRouterID,
+		LinkedLogicalSwitchPortId: makeResourceReference("LogicalPort", linkedLogicalSwitchPortID),
+		EdgeClusterMemberIndex:    edgeClusterMemberIndex,
+		Subnets:                   subnets,
+		UrpfMode:                  urpfMode,
+		MacAddress:                macAddress,
+	}
+	logicalRouterUpLinkPort, resp, err := nsxClient.LogicalRoutingAndServicesApi.CreateLogicalRouterUpLinkPort(nsxClient.Context, logicalRouterUpLinkPort)
+
+	if err != nil {
+		return fmt.Errorf("Error during LogicalRouterUpLinkPort create: %v", err)
+	}
+
+	if resp != nil && resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("Unexpected status returned during LogicalRouterUpLinkPort create: %v", resp.StatusCode)
+	}
+
+	d.SetId(logicalRouterUpLinkPort.Id)
+
+	return resourceNsxtLogicalRouterUpLinkPortRead(d, m)
+}
+
+func resourceNsxtLogicalRouterUpLinkPortRead(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(nsxtClients).NsxtClient
+	if nsxClient == nil {
+		return resourceNotSupportedError()
+	}
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining logical router uplink port id")
+	}
+
+	logicalRouterUpLinkPort, resp, err := nsxClient.LogicalRoutingAndServicesApi.ReadLogicalRouterUpLinkPort(nsxClient.Context, id)
+	if resp != nil && resp.StatusCode == http.StatusNotFound {
+		log.Printf("[DEBUG] LogicalRouterUpLinkPort %s not found", id)
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error during LogicalRouterUpLinkPort read: %v", err)
+	}
+
+	d.Set("revision", logicalRouterUpLinkPort.Revision)
+	d.Set("description", logicalRouterUpLinkPort.Description)
+	d.Set("display_name", logicalRouterUpLinkPort.DisplayName)
+	setTagsInSchema(d, logicalRouterUpLinkPort.Tags)
+	d.Set("logical_router_id", logicalRouterUpLinkPort.LogicalRouterId)
+	d.Set("linked_logical_switch_port_id", logicalRouterUpLinkPort.LinkedLogicalSwitchPortId.TargetId)
+	d.Set("edge_cluster_member_index", logicalRouterUpLinkPort.EdgeClusterMemberIndex)
+	d.Set("subnets", logicalRouterUpLinkPort.Subnets)
```

Comment:
This is a complex type, so a setter function that converts sdk struct to schema map is required (please refer to setTagsInSchema as an example)